core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -19,11 +19,38 @@
 
 package org.apache.iceberg;
 
+import java.util.Set;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+
 public class TableProperties {
 
   private TableProperties() {
   }
 
+  /**
+   * Reserved table property for table format version.
+   * <p>
+   * Iceberg will default a new table's format version to the latest stable and recommended version.
+   * This reserved property keyword allows users to override the Iceberg format version of the table metadata.
+   * <p>
+   * If this table property exists when creating a table, the table will use the specified format version.
+   * If a table updates this property, it will try to upgrade to the specified format version.
+   * This helps developers to try out new features in an unreleased version or migrate existing tables to a new version.
```

Comment:
Nit: this sentence conflicts with the note that the property isn't used for incomplete or unstable versions.